### ISSUE TITLE
OPTICKS-1229 - UndoGroup and UndoLock should reset the View pointer when the view is destroyed

### ISSUE THREAD

**Issue by tclarke** [opticks-org/opticks#257]
Both UndoGroup and UndoLock classes store a View pointer that is used to close an undo group or unblock undo actions when the object is destroyed. Both classes should store the View pointer as a SafePtr so that the pointer is automatically set to NULL if the view is destroyed. Currently, if the view is destroyed before the UndoGroup or UndoLock object is destroyed, the application will crash.
                          
Original Issue: [OPTICKS-1229](https://issues.opticks.org/jira/browse/OPTICKS-1229)
Status: Deferred
Feature Interpretation:

    
Suggested Solution:

Actual Solution:

Severity: 
Version Control: 
Environment: 
Linked:
